sw/device/tests/sim_dv/flash_escalation_reset_test.c

```diff
@@ -131,24 +202,32 @@ static void alert_handler_config(void) {
 
   dif_alert_handler_escalation_phase_t esc_phases[] = {
       {.phase = kDifAlertHandlerClassStatePhase0,
-       .signal = 0,
+       .signal = 0xFFFFFFFF,  // do not trigger any signal, just wait.
        .duration_cycles =
-           alert_handler_testutils_get_cycles_from_us(kEscalationPhase0Micros) *
-           alert_handler_testutils_cycle_rescaling_factor()},
+           alert_handler_testutils_get_cycles_from_us(kEscalationPhase0Micros)},
       {.phase = kDifAlertHandlerClassStatePhase1,
-       .signal = 3,
+       .signal = 0,  // NMI
```

Comment:
nit: are these constants published anywhere, e.g. in an autogenerated file? A comment to a doc would help too.

Reply:
I copied from [here](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/tests/sim_dv/all_escalation_resets_test.c;drc=cc31840bc26892d7be406972aa1868853d8e6660;l=652)
Not sure if this can be autogenerated but at least we can document for other users.
I will create an issue for this and work with @matutem

Reply:
I create #16894 to follow up this.

Reply:
This is not available in the autogen'd headers.
The UINT32_MAX value has been added as a workaround for a DIF limitation:
https://github.com/lowRISC/opentitan/blob/6822f12051695de4464988b158b98dd48660fb32/sw/device/lib/dif/dif_alert_handler.c#L344-L374

I.e., the DIF assumed that exactly one escalation signal is being driven in each phase, whereas the alert handler actually supports the full assignment matrix. The UINT32_MAX value has been added to signify an empty assignment matrix in this case - but we should probably refactor this as stated in the comment in the code.

Reply:
It seems this dif_alert_handler code deserves a priority 1 issue. Also, I think we need to document the meaning of each signal (0..3) since they are wired in the hardware: the relation of signal and phase is programmable, but signals are fixed.

The description in https://docs.opentitan.org/hw/top_earlgrey/ip_autogen/alert_handler/doc/#programmable-escalation-protocol is not sufficient for top_earlgrey since it ignores the specific connection of the signals. Where is it (or should it be) described?